api/core/v1alpha1/common_types.go

```diff
@@ -191,3 +203,113 @@ type ExternalLogSet struct {
 	// +required
 	HAKeeperEndpoint string `json:"haKeeperEndpoint,omitempty"`
 }
+
+type CloneSetScaleStrategy struct {
+	// PodsToDelete is the names of Pod should be deleted.
+	// Note that this list will be truncated for non-existing pod names.
+	// +optional
+	PodsToDelete []string `json:"podsToDelete,omitempty"`
+
+	// The maximum number of pods that can be unavailable during the update.
+	// Value can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%).
+	// Absolute number is calculated from percentage by rounding down.
+	// Defaults to 25%.
+	// +optional
+	// +kubebuilder:default="25%"
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+}
+
+type CloneSetUpdateStrategy struct {
+	// Type indicates the type of the CloneSetUpdateStrategy.
+	// Default is ReCreate.
+	// +optional
+	// +kubebuilder:default="ReCreate"
+	Type kruise.CloneSetUpdateStrategyType `json:"type,omitempty"`
```

Comment:
`v1alpha1` should not expose API of `CloneSet`, since `CloneSet` is an implementation detail and may be changed in the future. We may expose `PodSetUpdateStrategy` and map it the implementation instead.